extensions/elytron-security-ldap/runtime/src/main/java/io/quarkus/elytron/security/ldap/LdapRecorder.java

```diff
@@ -0,0 +1,79 @@
+package io.quarkus.elytron.security.ldap;
+
+import java.security.Provider;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+import javax.naming.NamingException;
+import javax.naming.directory.DirContext;
+
+import org.wildfly.common.function.ExceptionSupplier;
+import org.wildfly.security.WildFlyElytronProvider;
+import org.wildfly.security.auth.realm.ldap.AttributeMapping;
+import org.wildfly.security.auth.realm.ldap.DirContextFactory;
+import org.wildfly.security.auth.realm.ldap.LdapSecurityRealmBuilder;
+import org.wildfly.security.auth.realm.ldap.SimpleDirContextFactoryBuilder;
+import org.wildfly.security.auth.server.SecurityRealm;
+
+import io.quarkus.elytron.security.ldap.config.AttributeMappingConfig;
+import io.quarkus.elytron.security.ldap.config.DirContextConfig;
+import io.quarkus.elytron.security.ldap.config.IdentityMappingConfig;
+import io.quarkus.elytron.security.ldap.config.LdapSecurityRealmConfig;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class LdapRecorder {
+
+    private static final Provider[] PROVIDERS = new Provider[] { new WildFlyElytronProvider() };
+
+    /**
+     * Create a runtime value for a {@linkplain LdapSecurityRealm}
+     *
+     * @param config - the realm config
+     * @return - runtime value wrapper for the SecurityRealm
+     */
+    public RuntimeValue<SecurityRealm> createRealm(LdapSecurityRealmConfig config) {
+        Supplier<Provider[]> providers = () -> PROVIDERS;
```

Comment:
We try to avoid lambda inside Quarkus extension code

Reply:
Reverted to anonymous function instead of lambda.